crates/hotshot/src/traits/networking/push_cdn_network/definition/message_hook.rs

```diff
@@ -0,0 +1,482 @@
+#![allow(clippy::unnecessary_wraps)]
+use std::hash::Hasher;
+use std::num::NonZeroUsize;
+use std::sync::atomic::{AtomicU64, Ordering};
+use std::sync::Arc;
+use std::time::Instant;
+
+use anyhow::{Context, Result};
+use cdn_broker::reexports::def::hook::{HookResult, MessageHookDef};
+use cdn_broker::reexports::message::{Broadcast, Direct, Message as PushCdnMessage};
+use lru::LruCache;
+use parking_lot::Mutex;
```

Comment:
is this better performance than async_std mutex? just curious why we use this mutex

Reply:
Yeah, related comment [here](https://github.com/EspressoSystems/HotShot/pull/3671#discussion_r1757538068). I can just go with the `std` mutex if we need

Reply:
no need more out of curiosity